### PR TITLE
2.0.0 rc rebase 2

### PR DESCRIPTION
**Pull request by ktangsali**
# PhysicsNeMo Pull Request

## Description
2nd Rebase of RC Branch into Main

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py [hunk 1]
```diff
@@ -304,10 +304,10 @@ def build_data_dict(
...
     data_dict["air_density"] = torch.tensor(
-        [air_density], device=device, dtype=torch.float32
+        air_density, device=device, dtype=torch.float32
     )
     data_dict["stream_velocity"] = torch.tensor(
-        [stream_velocity], device=device, dtype=torch.float32
+        stream_velocity, device=device, dtype=torch.float32
     )
 
     return data_dict
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py:313]: Out of scope of this PR, since this is just a rebase.

File: physicsnemo/diffusion/guidance/__init__.py [hunk 2]
```diff
@@ -18,14 +18,14 @@
...
     ModelConsistencyDPSGuidance,
 )
 
 __all__ = [
     "DPSGuidance",
-    "DPSDenoiser",
+    "DPSScorePredictor",
     "ModelConsistencyDPSGuidance",
     "DataConsistencyDPSGuidance",
 ]
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply physicsnemo/diffusion/guidance/__init__.py:31]: Out of scope of this PR, since this is just a rebase.

File: physicsnemo/diffusion/guidance/dps_guidance.py [hunk 3]
```diff
@@ -390,6 +388,14 @@ def __call__(
             Guided score of same shape :math:`(B, *)` as ``x``. Computed as the
             sum of the unconditional score and all guidance terms.
         """
+        if not torch.compiler.is_compiling() and torch.is_inference_mode_enabled():
+            raise RuntimeError(
+                "DPSScorePredictor requires autograd but torch inference mode "
+                "is enabled. Wrap the calling code with "
+                "'with torch.inference_mode(False):' or 'with torch.no_grad():' "
+                "instead."
+            )
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply physicsnemo/diffusion/guidance/dps_guidance.py:397]: Out of scope of this PR, since this is just a rebase.
